test/test_c10d.py

```diff
@@ -0,0 +1,186 @@
+import math
+import multiprocessing
+import sys
+import tempfile
+import unittest
+from functools import wraps
+
+import torch
+import torch.c10d as c10d
+
+from common import TestCase
+
+
+TCP_ADDR = '127.0.0.1'
+TCP_PORT = 29500
+
+TIMEOUT_DEFAULT = 5
+TIMEOUT_OVERRIDE = {}
+
+
+def get_timeout(test_id):
+    return TIMEOUT_OVERRIDE.get(test_id.split('.')[-1], TIMEOUT_DEFAULT)
+
+
+if not c10d.is_available():
+    print('c10d not available, skipping tests')
+    sys.exit(0)
+
+
+class StoreTestBase(object):
+    def _create_store(self, i):
+        raise RuntimeError("implement this")
+
+    def _test_set_get(self, fs):
+        fs.set("key0", "value0")
+        fs.set("key1", "value1")
+        fs.set("key2", "value2")
+        self.assertEqual(b"value0", fs.get("key0"))
+        self.assertEqual(b"value1", fs.get("key1"))
+        self.assertEqual(b"value2", fs.get("key2"))
+
+    def test_set_get(self):
+        self._test_set_get(self._create_store())
+
+
+class FileStoreTest(TestCase, StoreTestBase):
+    def setUp(self):
+        self.file = tempfile.NamedTemporaryFile()
```

Comment:
You need to delete the file as well.

Reply:
No need, from the docs: "If delete is true (the default), the file is deleted as soon as it is closed."

Reply:
Well you never call `.close()` and I'm pretty sure it's not closed automatically in `__del__`. Those objects are usualy used as context managers, and are deleted once you exit the scope.